### PR TITLE
Removing readonly for name

### PR DESCRIPTION
**Pull request by amolr**
Removed readonly for name parameter

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@amolr__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (amolr). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-notificationhubs/2016-03-01/swagger/notificationhubs.json [hunk 1]
```diff
@@ -1819,7 +1819,6 @@
           "description": "Resource Id"
         },
         "name": {
-          "readOnly": true,
```
Comment (johanste) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: Why are you removing the readOnly declaration for the resource definition?

Comment (amolr) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: We are using name as part of the request payload which needs serialization.

Comment (johanste) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: ..but are you accepting a **Resource** in your payload? Conceptually, you are changing how your resources are modeled (which affects **all** usages or Resource) in order to satisfy the serialization requirements for one api.

Comment (amarzavery) [inline arm-notificationhubs/2016-03-01/swagger/notificationhubs.json]: @amolr - Another way to isolate this problem is to define all the properties for the body parameter of checkNamespaceAvailability and it should not inherit from Resource. Then you can keep name as a "readonly" property on Resource.

**Pull request merged**